specs/lightclient/beacon-chain.md

```diff
@@ -0,0 +1,208 @@
+    """
+    return (
+        ValidatorIndex(compact_validator >> 16),
+        bool((compact_validator >> 15) % 2),
+        compact_validator & (2**15 - 1),
+    )
+```
+
+#### `committee_to_compact_committee`
+
+```python
+def committee_to_compact_committee(state: BeaconState, committee: Sequence[ValidatorIndex]) -> CompactCommittee:
+    """
+    Given a state and a list of validator indices, outputs the ``CompactCommittee`` representing them.
+    """
+    validators = [state.validators[i] for i in committee]
+    compact_validators = [
+        pack_compact_validator(i, v.slashed, v.effective_balance // EFFECTIVE_BALANCE_INCREMENT)
+        for i, v in zip(committee, validators)
+    ]
+    pubkeys = [v.pubkey for v in validators]
+    return CompactCommittee(pubkeys=pubkeys, compact_validators=compact_validators)
+```
+
+### Beacon state accessors
+
+#### `get_light_client_committee`
+
+```python
+def get_light_client_committee(beacon_state: BeaconState, epoch: Epoch) -> Sequence[ValidatorIndex]:
+    """
+    Return the light client committee of no more than ``LIGHT_CLIENT_COMMITTEE_SIZE`` validators.
+    """
+    source_epoch = (max(epoch // LIGHT_CLIENT_COMMITTEE_PERIOD, 1) - 1) * LIGHT_CLIENT_COMMITTEE_PERIOD
+    active_validator_indices = get_active_validator_indices(beacon_state, source_epoch)
+    seed = get_seed(beacon_state, source_epoch, DOMAIN_LIGHT_CLIENT)
+    return [
+        compute_shuffled_index(i, active_validator_indices, seed)
+        for i in range(min(active_validator_indices, LIGHT_CLIENT_COMMITTEE_SIZE))
+    ]
+```
+
+### Block processing
+
+```python
+def process_block(state: BeaconState, block: BeaconBlock) -> None:
+    phase0.process_block(state, block)
+    process_light_client_signature(state, block.body)
+```
+
+#### Light client processing
+
+```python
+def process_light_client_signature(state: BeaconState, block_body: BeaconBlockBody) -> None:
+    committee = get_light_client_committee(state, get_current_epoch(state))
+    previous_slot = max(state.slot, 1) - 1
+    previous_block_root = get_block_root_at_slot(state, previous_slot)
+
+    # Light clients sign over the previous block root
+    signing_root = compute_signing_root(
+        previous_block_root,
+        get_domain(state, DOMAIN_LIGHT_CLIENT, compute_epoch_at_slot(previous_slot))
+    )
+    
+    participants = [
+        committee[i] for i in range(len(committee)) if block_body.light_client_bits[i]
+    ]
+    
+    signer_pubkeys = [
+        state.validators[participant].pubkey for participant in participants
+    ]
+    
+    assert bls.FastAggregateVerify(signer_pubkeys, signing_root, block_body.light_client_signature)
+    
+    # Process rewards
+    total_reward = Gwei(0)
+    active_validator_count = len(get_active_validator_indices(beacon_state, get_current_epoch(state)))
+    for participant in participants:
+        reward = get_base_reward(state, participant) * active_validator_count // len(committee)
```

Comment:
do we want to further reduce the reward by `SLOTS_PER_EPOCH`? e.g.

```python
reward = ...
reward = reward // SLOTS_PER_EPOCH
```

`BASE_REWARDS_PER_EPOCH` has been increased to `5` but as it stands this reward is assessed per slot.

Reply:
and it *feels* like we also want to reduce the per-participant reward (proportionately) to account for the proposer reward... unless i'm missing something w/ the rewards rationale here.